(DOCSP-41504, DOCSP-41866, DOCSP-41583): Adds TOC label guidance, SEO, and spelling updates. (#175)
mongodb/docs-meta: source/style-guide/style/toc-labels.txt

.. _toc-labels:

========================
Table of Contents Labels
========================

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

Table of Contents (TOC) labels refer to the headings or titles used to 
represent the various sections or pages of our documentation in the 
TOC. The TOC provides an organized list of the documentation's 
structure and content.

Each entry in the TOC typically includes a label or heading that 
corresponds to a section or page in the documentation. These labels 
give users a quick overview of the topics covered and help them 
navigate to specific parts of the documentation more efficiently.

When you create a TOC, you can generate these labels automatically 
based on the titles of your pages, or specify labels that differ from 
the page title.

Page titles and Table of Contents (TOC) labels serve distinct purposes. 
We should craft them with different considerations.

TOC labels and page titles can match. However, we should define them 

> separately even when they match as a good habit and for added 

Reviewer: [Nit, minimalism]

I think we can drop "as a good habit" here.

```suggestion
separately even when they match for added 
```

flexibility. This approach ensures that we can easily implement any 
future changes without affecting the other element. Separate 
definitions for TOC labels and page titles contribute to organized 
content management and a seamless user experience.

While both page titles and TOC labels contribute to the overall user 
experience, page titles focus more on attracting users and SEO, while 
TOC labels prioritize representing the document's structure and aiding 
in navigation within the document itself.

Page Titles
-----------

- Optimize page titles for search engines. They should include relevant 
  keywords to improve the page's discoverability in search results. To 
  learn more, see :ref:`seo-guidelines`.

- Create succinct, meaningful, descriptive page titles, and place the 
  most important words first. Page titles should have a maximum of 70 
  characters and provide a quick overview of the content. Search 
  results or browser tabs might truncate extra long titles. Avoid 
  excessive or irrelevant words (keyword stuffing).

- While we should craft concise titles, include enough information to 
  confirm to the user that they have landed in the right place. A page 

>   title can provide a bit more detail than a TOC label.

Reviewer: [Nit, minimalism]

```suggestion
  title can provide more detail than a TOC label.
```

- Ensure that each title and heading is unique. Identical titles, even 
  between documentation sets, might compete in search results.

- Don't include "MongoDB" in a title unless the page is a product 
  landing page.

> - Aim to attract users and encourage them to click on the link.

Reviewer: [Suggestion]

To add a bit more context here, maybe we can expound on how to do this. For example:

```suggestion
- Aim to attract users and encourage them to click on the link. Include phrases users might search for, or tasks they might want to accomplish.
```

- Create page titles that are readable and effective on various 
  devices, including mobile. This involves considering character limits 
  and readability on smaller screens.

- Use a consistent grammatical structure, style, and tone across all 
  page titles to maintain a unified and professional appearance. For 
  example, use verbs to indicate procedural content.

To learn more, see :ref:`titles-and-headings`.

TOC Labels
----------

> - While the automatic labels make it easier for us to 

Reviewer: ```suggestion
- While the automatic labels make it easier for writers to 
```

  update the TOC as the content evolves, long titles interfere with 

>   efficient navigation. So, we recommend that you specify TOC labels to 

Reviewer: ```suggestion
  efficient navigation. We recommend that you specify TOC labels to 
```

  enhance the document's accessibility and usability.

- Craft clear and descriptive TOC labels that convey the content of 
  each section or page accurately. Users should understand the context 

>   from the TOC label and its placement in the overall TOC hierarchy.

Reviewer: [Praise]

I really like the mention of the overall hierarchy here!

> - Avoid line-wrapping. Let the placement of the TOC label in the 

Reviewer: [Question]

Just making sure I'm reading this correctly, this is saying that TOC labels should appear on a single line of text? If so, maybe we can consider framing this as what _to_ do, rather than what _not to_ do:

```suggestion
- Ensure your TOC label fits on a single line of text. Let the placement of the TOC label in the 
```

-----------

- Optimize page titles for search engines. They should include relevant 
  keywords to improve the page's discoverability in search results. To 
  learn more, see :ref:`seo-guidelines`.

- Create succinct, meaningful, descriptive page titles, and place the 
  most important words first. Page titles should have a maximum of 70 
  characters and provide a quick overview of the content. Search 
  results or browser tabs might truncate extra long titles. Avoid 
  excessive or irrelevant words (keyword stuffing).

- While we should craft concise titles, include enough information to 
  confirm to the user that they have landed in the right place. A page 
  title can provide a bit more detail than a TOC label.

- Ensure that each title and heading is unique. Identical titles, even 
  between documentation sets, might compete in search results.

- Don't include "MongoDB" in a title unless the page is a product 
  landing page.

- Aim to attract users and encourage them to click on the link.

- Create page titles that are readable and effective on various 
  devices, including mobile. This involves considering character limits 
  and readability on smaller screens.

> - Use a consistent grammatical structure for page titles. Use a 
>   consistent style and tone across all page titles to maintain a 
>   unified and professional appearance. For example, use verbs to 
>   indicate procedural content.

Reviewer 2: ```suggestion
- Use a consistent grammatical structure, style, and tone across all page titles to 
  maintain a unified and professional appearance. For example, use verbs to 
  indicate procedural content.
```
2 sentences could be combined for conciseness and clarity.

  enhance the document's accessibility and usability.

- TOC labels primarily represent the documentation's structure. They 
  outline the hierarchy of sections and help users navigate through the 
  content.

- Craft clear and descriptive TOC labels that convey the content of 
  each section or page accurately. Users should understand the context 
  from the TOC label and its placement in the overall TOC hierarchy.

- Avoid line-wrapping. Let the placement of the TOC label in the 
  overall TOC hierarchy provide context. Avoid redundancy and 
  repetition. The page title itself can provide a bit more detail and 
  confirm to the user that they have landed in the right place. To this 
  end, you can use industry standard abbreviations and the ampersand 
  when needed.

  .. important::

     In favor of brevity, TOC labels do allow for the the ampersand ( & ) symbol.

- Don't stray so far from the original page title that you disorient 
  the user.

- TOC labels should follow a logical order that reflects the 
  documentation's flow. You might include numbering or an order that 
  makes sense for the documentation's purpose.

> - TOC labels primarily aid user navigation. Users should be able to 
>   quickly locate and go to specific sections of interest.

Reviewer 2: ```suggestion

```
Think this point could be removed or added on to the second bullet, as they say similar things: "TOC labels primarily represent the documentation's structure. They outline the hierarchy of sections and help users navigate through the 
content."
